### PR TITLE
[mds-agency] [mds-web-sockets] Improve agency/telemetry error handling, and add basic health check to WebSocket client

### PR DESCRIPTION
**Pull request by avatarneil**
Previously, agency/telemetry POST requests would send a 400 response to clients on a failed write to cache/stream/socket; this PR resolves the extraneous error bubbling to the API client. It also adds a basic health check to the internal WebSocket client used by agency to validate that the WebSocket http server is running before attempting to establish a connection.

## PR Checklist

 - [ ] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [ ] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [x] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

**Comment by avatarneil** [comment]
Tested in an environment with `mds-web-sockets` running, and an environment without. When not running the web-socket client, we simply log the error and move on!

File: packages/mds-agency/request-handlers.ts [hunk 1]
```diff
@@ -412,12 +412,6 @@ export const submitVehicleTelemetry = async (req: AgencyApiRequest, res: AgencyA
     if (valid.length) {
       const recorded_telemetry = await writeTelemetry(valid)
 
-      try {
```
Comment (macsj200) [inline packages/mds-agency/request-handlers.ts]: Was this a redundant call?

File: packages/mds-web-sockets/client.ts [hunk 2]
```diff
@@ -13,29 +14,43 @@ async function sendAuth() {
...
-  connection.onerror = async err => {
-    return log.error(err)
-  }
+    setWsHeartbeat(connection as WebSocketBase, 'PING')
+
+    connection.onopen = async () => {
+      await sendAuth()
+    }
+
+    connection.onerror = async err => {
```
Comment (macsj200) [inline packages/mds-web-sockets/client.ts]: I know this was already here, but I don't think this function needs an `async` modifier

File: packages/mds-agency/request-handlers.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-agency/request-handlers.ts]: Yup

**Pull request merged**